### PR TITLE
Copy the same technique from Nomad re: not recompiling tests

### PR DESCRIPTION
**Pull request by sean-**

### CODE REVIEW

File: scripts/test.sh [hunk 1]
```diff
@@ -12,4 +12,9 @@ go build -o $TEMPDIR/consul || exit 1
 
 # Run the tests
 echo "--> Running tests"
-go list ./... | grep -v '^github.com/hashicorp/consul/vendor/' | PATH=$TEMPDIR:$PATH xargs -n1 go test ${GOTEST_FLAGS:--cover -timeout=360s}
+GOBIN="`which go`"
+go list ./... | grep -v '^github.com/hashicorp/consul/vendor/' | \
+    sudo \
```
Comment (slackpad) [inline scripts/test.sh]: I don't think we need sudo.

Comment (sean-) [inline scripts/test.sh]: Yeah, that bombed out... Nomad needs it, not here.  _waiting for the tests to complete_

**Comment by slackpad** [comment]
@sean- any thoughts on this one? I don't fully understand what the difference is here.

**Comment by slackpad** [comment]
We should snag the incidental port numbering change at least, tho.

**Comment by slackpad** [comment]
Similar changes came in around https://github.com/hashicorp/consul/commit/0a04d23f505d41fdeff338f671734efde6a717b7 so closing this out.

**Pull request closed without merging**